File: python/51-100/71. Simplify Path.py
class Solution(object):
    def simplifyPath(self, path):
        """
        :type path: str
        :rtype: str
        """
        path = path.split("/")
        path = filter(lambda x : x != "", path)
        res = []
        for val in path:
            if val != "." and val != "..":
                res.append(val)
            else:
                if val == ".":
                    continue
                if val == "..":
                    if res:
                        res.pop()
        return "/" + "/".join(res)

File: cpp/51-100/71. Simplify Path.cpp
class Solution {
public:
	string simplifyPath(string path)
	{
		vector<string> directory;
		string res = "", temp;
		stringstream ss(path);
		while(getline(ss, temp, '/')){
			if(temp == "." || temp == "")
				continue;
			if(temp == ".." && !directory.empty())
				directory.pop_back();
			if(temp != "..")
				directory.push_back(temp);
		}
		for(int i = 0 ; i < directory.size(); i++)
			res += "/" + directory[i];
		return res.empty()? "/":res;
	}
};